build.py

```diff
@@ -402,22 +402,9 @@ def core_cmake_args(components, backends, install_dir):
                           in FLAGS.filesystem))
 
     cargs.append(
-        cmake_core_enable('TRITON_ENABLE_TENSORFLOW',
-                          ('tensorflow1' in backends) or
-                          ('tensorflow2' in backends)))
-
-    for be in (CORE_BACKENDS + NONCORE_BACKENDS):
-        if not be.startswith('tensorflow'):
-            cargs.append(
-                cmake_core_enable('TRITON_ENABLE_{}'.format(be.upper()), be
-                                  in backends))
-        if be == 'tensorrt':
-            cargs += tensorrt_cmake_args()
-        if (be in CORE_BACKENDS) and (be in backends):
-            if be == 'ensemble':
-                pass
-            else:
-                fail('unknown core backend {}'.format(be))
+        cmake_core_enable('TRITON_ENABLE_ENSEMBLE', 'ensemble' in backends))
+    cargs.append(
+        cmake_core_enable('TRITON_ENABLE_TENSORRT', 'tensorrt' in backends))
```

Comment:
Don't we still need to add the backend flags to core for supporting old model config (only specifies "platform" but not "backend") https://github.com/triton-inference-server/core/blob/main/src/model_config_utils.cc#L792-L794

Reply:
I just made that part of the core be always be included. If a model specifies and old platform that isn't actually available then it will fail because the backend won't be available.

Reply:
The only reason TRITON_ENABLE_TENSORRT is there is because we need to know when to build caffe2plan or not.